crates/turborepo-lib/src/shim.rs

```diff
@@ -176,50 +177,237 @@ struct PackageJson {
     version: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct LocalTurboState {
-    bin_path: PathBuf,
-    version: String,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct YarnRc {
+    pnp_unplugged_folder: PathBuf,
 }
 
-impl LocalTurboState {
-    pub fn infer(repo_root: &Path) -> Option<Self> {
-        let local_turbo_path = repo_root.join("node_modules").join(".bin").join({
+impl Default for YarnRc {
+    fn default() -> Self {
+        Self {
+            pnp_unplugged_folder: PathBuf::from("").join(".yarn").join("unplugged"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TurboState {
+    bin_path: Option<PathBuf>,
+    version: &'static str,
+    repo_state: Option<RepoState>,
+}
+
+impl Default for TurboState {
+    fn default() -> Self {
+        Self {
+            bin_path: env::current_exe().ok(),
+            version: get_version(),
```

Comment:
should this be `Self::version()`?

Reply:
That's entirely unused at this step but I've got it lined up for the third PR in this stack, but I bit off too much and walked it back. To avoid rebasing through the PR stack I want to leave it here for now.